internal/expr/parser.go

```diff
@@ -473,35 +547,106 @@ func (p *Parser) parseColumn() (columnAccessor, bool, error) {
 	return columnAccessor{}, false, nil
 }
 
-func (p *Parser) parseTargetType() (valueAccessor, bool, error) {
+func (p *Parser) parseTargetType() (memberAcessor, bool, error) {
+	startLine := p.lineNum
+	startCol := p.colNum()
+
 	if p.skipByte('&') {
+		// Using a slice as an output is an error, we add the case here to
+		// improve the error message.
+		if st, ok, err := p.parseSliceAccessor(); err != nil {
+			return memberAcessor{}, false, err
+		} else if ok {
+			return memberAcessor{}, false, errorAt(fmt.Errorf("cannot use slice syntax %q in output expression", st), startLine, startCol, p.input)
+		}
 		return p.parseTypeName()
 	}
 
-	return valueAccessor{}, false, nil
+	return memberAcessor{}, false, nil
+}
+
+// parseUNumber parses a non-negative number composed of one or more digits.
+func (p *Parser) parseUNumber() (uint64, bool) {
+	mark := p.pos
+	if !p.skipUNumber() {
+		return 0, false
+	}
+	n, err := strconv.ParseUint(p.input[mark:p.pos], 10, 64)
+	if err != nil {
+		panic("internal error: skipUNumber did not skip a valid number")
```

Comment:
We don't want to panic in a library. This could be quite disruptive for a Juju controller for example.

Reply:
In the initial implementation I was just ignoring the error because it meant that `skipUNumber` was failing internally, but after a few revisions of this PR I changed it to a panic to be explicit about why we were ignoring the error. The other option would be to return `(int, bool, error)` which seems a little bit overkill. What do you think is the best approach?